### ISSUE TITLE
New Rule: (Content Security Policy (CSP) directive inspection)

### ISSUE THREAD

**Issue by pubmikeb** [eslint/eslint#15101]
### Rule details

The rule should check the correctness of the CSP directives spelling

### Related ECMAScript feature

Content Security Policy (CSP)

### What type of rule is this?

Warns about a potential problem

### Example code
Given a wrongly spelled CSP directive:
```js
image-src 'self';
```

A correctly spelled CSP directive:
```js
img-src 'self';
```

### Why should this rule be in the core instead of a plugin?

Playing with the Content Security Policy (CSP) inspections I paid attention that I used a wrong `image-src` instead of the correct `img-src`, it would be great if ESLint could detect a wrong CSP directive and suggest a correct one instead.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

Such rule could save a lot of time since for a human eye it might be difficult to spot the difference between `image-src` and `img-src`.

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
